### ISSUE TITLE
the jupyter notebook of pwcnet_finetune_sm-6-2-cyclic-chairsthingsmix.ipynb has some errors

### ISSUE THREAD

**Issue by jsczzzk** [philferriere/tfoptflow#7]
There are some errors at the end of this jupyter notebook like this ''2018-09-13 22:47:41 Iter 7000 [Train]: loss=nan, epe=nan,''

**Comment by philferriere** [comment]
As you train the PWC-Net models, you will notice that we sometimes hit numeric instability (i.e., NaNs). The current training code requires a fair amount of babysitting, relying on you to restart the training when/if the training cannot recover from those NaNs. 

An alternate approach is to modify the training code so that when NaNs are detected, no gradient update is performed. To do this, replace calls to the optimizer's `minimize()` method with two sequential calls to `compute_gradients()` and `apply_gradients()`, making sure to test the resulting gradients in-between. 

Hope thils helps!

**Comment by jeffbaena** [comment]
I am now testing if it works, but an easy fix as proposed by phil could be adding these  lines from line 540 of model_pwcnet.py

I noticed this code might not be optimal as I see a lot of cpu computation and low GPU volatile memory utilization. I batch size of 1000 samples takes around 2,5 hours with a volta v100 16 gb, magnetic memory and 16 cpu cores. Could anyone help?

            grads_and_vars = self.optim.compute_gradients(self.loss_op, var_list=tf.trainable_variables())
            if tf.is_nan(grads_and_vars[0]) == True:
                grads_and_vars_ = [(tf.where(tf.is_nan(grad), tf.zeros_like(grad), grad), val) for grad, val in
                                   grads_and_vars]
            elif tf.is_nan(grads_and_vars[1]) == True:
                grads_and_vars_ = [(tf.where(tf.is_nan(grad), tf.zeros_like(grad), grad), val) for grad, val in
                                   grads_and_vars]
            else:
                grads_and_vars_ = grads_and_vars
            self.optim_op = self.optim.apply_gradients(grads_and_vars_, global_step=self.g_step_op, 
            name=None)

**Comment by lelelexxx** [comment]
> I am now testing if it works, but an easy fix as proposed by phil could be adding these lines from line 540 of model_pwcnet.py
> 
> I noticed this code might not be optimal as I see a lot of cpu computation and low GPU volatile memory utilization. I batch size of 1000 samples takes around 2,5 hours with a volta v100 16 gb, magnetic memory and 16 cpu cores. Could anyone help?
> 
> ```
>         grads_and_vars = self.optim.compute_gradients(self.loss_op, var_list=tf.trainable_variables())
>         if tf.is_nan(grads_and_vars[0]) == True:
>             grads_and_vars_ = [(tf.where(tf.is_nan(grad), tf.zeros_like(grad), grad), val) for grad, val in
>                                grads_and_vars]
>         elif tf.is_nan(grads_and_vars[1]) == True:
>             grads_and_vars_ = [(tf.where(tf.is_nan(grad), tf.zeros_like(grad), grad), val) for grad, val in
>                                grads_and_vars]
>         else:
>             grads_and_vars_ = grads_and_vars
>         self.optim_op = self.optim.apply_gradients(grads_and_vars_, global_step=self.g_step_op, 
>         name=None)
> ```
Did you find the reason why this code is not efficient ? I have meet the similar problem using v100 16gb.
